scripts/get_objf_and_derivs_split.py

```diff
@@ -351,6 +354,57 @@ def MergeAndComputeObjfForSplit(split_index):
     MergeAllOrders(split_index)
     ComputeObjfAndFinalDerivs(split_index, args.derivs_out != None)
 
+def CleanUpIfNeeded():
+    # For cleaning up intermediate files that won't be used, note if option need-model = true
+    # be used, clean up will keep file 'float.all'
+    if args.clean_up == 'true':
+        try:
+            if not os.path.isdir(args.work_dir):
+                ExitProgram("error finding working directory {0}".format(args.work_dir))
+            # If need_model option set to be false, this script will keep "float.all" in
+            # each split dir for future use
+            need_split_float = True if args.need_model == 'false' else False
+            for split_index in range(1, args.num_splits + 1):
```

Comment:
this logic doesn't seem right to me.  Typically when this script is called, it's just get the derivatives w.r.t. some scalar values, so the split float.all files won't be used.  
You should probably add a separate boolean command-line argument --need-split-model which can be set from make_lm_dir.py in the case where it needs the split model.